src/Components/SmartComponents/HybridInventory/ImmutableDevicesTab/ImmutableDevices.js

```diff
@@ -96,56 +96,62 @@ const ImmutableDevices = ({ intl, cveName, filterRuleValues, inventoryRef, heade
         navigate(`/${systemId}?appName=vulnerabilities`);
     };
 
-    return isLoadingInventory ? <Spinner centered /> :
-        error?.hasError && !canReadHostsInventory    ? <ErrorHandler code={error?.errorCode} />
-            : (<AsynComponent
-                appName="inventory"
-                ref={inventoryRef}
-                module="./ImmutableDevices"
-                fallback={<div />}
-                store={ReducerRegistry.getStore()}
-                onLoad={({ mergeWithEntities }) => {
-                    ReducerRegistry.register({
-                        ...mergeWithEntities(
-                            inventoryEntitiesReducer([]),
-                            {
-                                page: Number(parameters.page || 1),
-                                perPage: DEFAULT_PAGE_SIZE,
-                                ...(parameters.sort && {
-                                    sortBy: {
-                                        key: parameters.sort.replace(/^-/, ''),
-                                        direction: parameters.sort.match(/^-/) ? 'desc' : 'asc'
-                                    }
-                                })
-                            }
-                        )
-                    });
-                }}
-                key="immutableDevices"
-                customFilters={{
-                    edgeParams: {
-                        ...parameters,
-                        host_type: 'edge'
-                    }
-                }}
-                getEntities={getEntities}
-                showActions={totalItems !== 0}
-                hideFilters={{ all: true, hostGroupFilter: false, operatingSystem: false }}
-                noSystemsTable={<EmptyStateNoSystems />}
-                mergeAppColumns={mergeAppColumns}
-                filterConfig={{
-                    items: [
-                        searchFilter,
-                        securityRuleFilter,
-                        advisoryNameFilter
-                    ]
-                }}
-                activeFiltersConfig={activeFilters}
-                onRowClick={onRowClick}
-            />
-            );
+    if (!isLoadingInventory && (error?.hasError || !canReadHostsInventory)) {
```

Comment:
Is this change needed for the tests?

Reply:
hmm, this is needed to show the Unauthorised error handler when RBAC returns false. Should I open a separate PR?